File: examples/ruby/keyring.rb
#!/usr/bin/ruby

require 'zypp'
include Zypp


keyring = ZYppFactory::instance.get_zypp.key_ring

path = Pathname.new("/suse/aschnell/tmp/repodata/repomd.xml.key")
puts path

publickey = PublicKey.new(path)
puts publickey

id = publickey.id()

puts "is key known/trusted #{keyring.is_key_known(id)} #{keyring.is_key_trusted(id)}"

keyring.import_key(publickey, true)

puts "is key known/trusted #{keyring.is_key_known(id)} #{keyring.is_key_trusted(id)}"

puts "list of known keys:"
keyring.public_keys.each do |key|
    puts key
end

puts "list of trusted keys:"
keyring.trusted_public_keys.each do |key|
    puts key
end

File: examples/python/keyring.py
#!/usr/bin/python

# test loading of pywsman
import sys

# cmake build dir
sys.path.insert(0, '../../build/swig/python')

from zypp import ZYppFactory, Pathname, KeyRing, PublicKey


keyring = ZYppFactory.instance().getZYpp().keyRing()

path = Pathname("/suse/aschnell/tmp/repodata/repomd.xml.key")
print path

publickey = PublicKey(path)
print publickey

id = publickey.id()

print "is key known/trusted %s %s" % (keyring.isKeyKnown(id), keyring.isKeyTrusted(id))

keyring.importKey(publickey, True)

print "is key known/trusted %s %s" % (keyring.isKeyKnown(id), keyring.isKeyTrusted(id))

print "list of known keys:"
for key in keyring.publicKeys():
    print key

print "list of trusted keys:"
for key in keyring.trustedPublicKeys():
    print key

